### DISCUSSION TITLE
Unable to log in with a second-level domain

### DISCUSSION THREAD

**Discussion by unhighghlow** [immich-app/immich#13672] [Q&A]
### The bug

The login button infinitely spins when trying to log in when using a second level domain (e.g. `sub.domain.tld`)

### The OS that Immich Server is running on

NixOS Unstable

### Version of Immich Server

v1.118.1

### Version of Immich Mobile App

none

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
none
```

### Your .env content

```Shell
none
```

### Reproduction steps

1. Set up an immich server
2. Point `theserver.org` and `immich.theserver.org` to the server's ip using /etc/hosts
3. Set up the admin account (works on both domains)
4. Log in on `theserver.org` (success)
5. Log in on `immich.theserver.org` (infinite spin, failure)

### Additional information

I was unable to test this on mobile, since I don't have access to the hosts file there

**Comment by bo0tzz** [comment]
Are there any errors in the log, the browser console, or the browser devtools network tab?

**Reply by unhighghlow** [reply]
The weird thing, there wasn't. The network tab showed /api/auth/login succeding!

**Reply by tbelway** [reply]
This is interesting... Can you tail the journal while logging in? Anything there?

**Reply by unhighghlow** [reply]
Here's the output of journalctl with IMMICH_LOG_LEVEL set to "verbose":
```
Oct 22 17:04:09 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:09 PM   DEBUG [Api:LoggingInterceptor~iqwt622u] GET /custom.css 200 25.69ms 192.168.100.75
Oct 22 17:04:10 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:10 PM   DEBUG [Api:LoggingInterceptor~dp3mvi46] GET /api/server/features 200 22.82ms 192.168.100.75
Oct 22 17:04:10 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:10 PM   DEBUG [Api:LoggingInterceptor~g40zmew9] GET /api/server/config 200 19.83ms 192.168.100.75
Oct 22 17:04:23 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:23 PM   DEBUG [Api:LoggingInterceptor~f9ihg71m] POST /api/auth/login 201 214.04ms 192.168.100.75
Oct 22 17:04:23 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:23 PM VERBOSE [Api:LoggingInterceptor~f9ihg71m] {"email":"admin@example.com","password":"********"}
Oct 22 17:04:23 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:23 PM   DEBUG [Api:GlobalExceptionFilter~mqv3do26] HttpException(401): {"message":"Authentication required","error":"Unauthorized","statusCode":401}
Oct 22 17:04:23 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:23 PM   DEBUG [Api:GlobalExceptionFilter~ryqu667q] HttpException(401): {"message":"Authentication required","error":"Unauthorized","statusCode":401}
Oct 22 17:04:23 nixos server[15152]: [Nest] 15152  - 10/22/2024, 5:04:23 PM   DEBUG [Api:GlobalExceptionFilter~85609aal] HttpException(401): {"message":"Authentication required","error":"Unauthorized","statusCode":401}
```

**Comment by bo0tzz** [comment]
What browser are you using?

**Reply by unhighghlow** [reply]
Firefox
